PennMobile/src/main/java/com/pennapps/labs/pennmobile/gsrFragment.java

```diff
@@ -0,0 +1,593 @@
+        int minutes = calendar.get(Calendar.MINUTE);
+        int hour = calendar.get(Calendar.HOUR);
+        int dayOfWeek = calendar.get(Calendar.DAY_OF_WEEK);
+        int day = calendar.get(Calendar.DAY_OF_MONTH);
+        int month = calendar.get(Calendar.MONTH) + 1;
+        int year = calendar.get(Calendar.YEAR);
+        int ampm = calendar.get(Calendar.AM_PM);
+
+
+        calendarButton.setText(year + "-" + month + "-" + day);
+
+
+        // Set default start/end times for GSR booking
+        String[] ampmTimes = getStartEndTimes(hour, minutes, ampm);
+        startButton.setText(ampmTimes[0]);
+        endButton.setText(ampmTimes[1]);
+
+         // Set up recycler view for list of GSR rooms
+         gsrRoomListRecylerView = (RecyclerView) v.findViewById(R.id.gsr_rooms_list);
+
+        /**
+         *
+         *
+         * START on click functions for buttons
+         *
+         */
+
+        //set start time button
+        startButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                // Get Current Time
+                final Calendar c = Calendar.getInstance();
+                int mHour = c.get(Calendar.HOUR_OF_DAY);
+                int mMinute = c.get(Calendar.MINUTE);
+
+                // Launch Time Picker Dialog
+                TimePickerDialog timePickerDialog = new TimePickerDialog(getActivity(),
+                        new TimePickerDialog.OnTimeSetListener() {
+
+                            @Override
+                            public void onTimeSet(TimePicker view, int hourOfDay,
+                                                  int minute) {
+                                String AM_PM ;
+                                if(hourOfDay < 12) {
+                                    AM_PM = "AM";
+                                } else {
+                                    AM_PM = "PM";
+                                }
+
+                                String hourString = Integer.toString(hourOfDay);
+                                if (hourOfDay == 0) {
+                                    hourString = "12";
+                                }
+                                //convert to nonmilitary time
+                                if (hourOfDay > 12) {
+                                    hourString = Integer.toString(hourOfDay - 12);
+                                }
+
+                                String minuteString = Integer.toString(minute);
+
+                                //Android treats minutes less than 10 as single digit
+                                if (minute < 10) {
+                                    minuteString = "0" + minute;
+                                }
+
+                                //display selected time
+                                startButton.setText(hourString + ":" + minuteString + " " + AM_PM);
+                            }
```

Comment:
Please use Joda time (or some other date/time parsing library). Do not write your own date formatter/parser.